merlin/models/data/ecommerce/aliccp/dataset.py

```diff
@@ -0,0 +1,431 @@
+import os
+import pickle
+import re
+import shutil
+from dataclasses import dataclass
+from glob import glob
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+from tqdm import tqdm
+
+import merlin.io
+from merlin.core.dispatch import get_lib
+from merlin.models.utils.example_utils import workflow_fit_transform
+from merlin.schema import Tags
+
+try:
+    import nvtabular as nvt
+
+    Workflow = nvt.Workflow
+except ImportError:
+    Workflow = None
+
+
+def get_aliccp(
+    path: Union[str, Path],
+    overwrite: bool = False,
+    transformed_name: str = "transformed",
+    nvt_workflow: Optional[Workflow] = None,
+    file_size: int = 100000,
+    **kwargs,
+) -> Tuple[merlin.io.Dataset, merlin.io.Dataset]:
+    """
+    Load the Ali-CCP: Alibaba Click and Conversion Prediction dataset.
+
+    We can't download this dataset automatically so this needs to be downloaded manually
+    To use this function, prepare by the following these 3 steps:
+    1. Download the raw data from
+    [tianchi.aliyun.com](https://tianchi.aliyun.com/dataset/dataDetail?dataId=408#1).
+    2. Unzip the raw data to a directory.
+    3. Run `prepare_alliccp(data_dir)` to convert the raw data to parquet files.
+
+    Downloading & preparing the data can take quite a while.
+    In case you want to use this dataset to run our tutorials, you can also opt for synthetic data.
+    Synthetic data can be generated by running::
+        from merlin.models.data.synthetic import generate_data
+
+        data = generate_data("aliccp")
+
+    Note, running this function requires the
+    [nvtabular](https://github.com/NVIDIA-Merlin/NVTabular) library.
+
+    Parameters
+    ----------
+    path: Union[str, Path]
+        Directory to load the raw data from.
+    overwrite: bool
+        Whether to overwrite the transformed data.
+    transformed_name: str
+        Name of the transformed data.
+        This is useful when you want to load the transformed data
+        multiple times using different workflows.
+    nvt_workflow: Optional[Workflow]
+        Workflow to transform the raw data.
+        If None, the default workflow will be used from `default_aliccp_transformation`.
+    file_size: int
+        Size of the parquet files to be loaded when preparing the data.
+
+    Returns
+    -------
+    train: merlin.io.Dataset
+        Training dataset.
+    valid: merlin.io.Dataset
+        Test dataset.
+    """
+
+    if not Workflow:
+        raise RuntimeError(
+            "Please install nvtabular in order to use this dataset",
+            "Run: `pip install nvtabular` to install it.",
```

Comment:
That’s not necesairily true (since you could pip install nvt in a conda-env that has rapids), but you do raise a good point! I guess we could detect whether the system has a GPU but doens’t have Rapids installed and therefore won’t be as fast as it could be. Let me add that!

Reply:
I added [this](https://github.com/NVIDIA-Merlin/models/blob/data/ali-ccp/merlin/models/utils/nvt_utils.py#L4) function that’s now used for the different datasets.